superset-frontend/src/explore/controlUtils.js

```diff
@@ -88,15 +88,17 @@ export const getControlConfig = memoizeOne(function getControlConfig(
   return control?.config || control;
 });
 
-export function applyMapStateToPropsToControl(control, state) {
-  if (control.mapStateToProps) {
-    const appliedControl = { ...control };
-    if (state) {
-      Object.assign(appliedControl, control.mapStateToProps(state, control));
-    }
-    return appliedControl;
+/**
+ * Call `mapStateToProps` from controlState and update it in place.
```

Comment:
is there a reason why we want to do mutation here? Unless it's going to have perf issues, i'd much prefer this function to return a new object and use immutable objects instead of mutating in place

Reply:
+1

Reply:
I thought there was merit to update this in place because once a state is updated, it should never be reused. Mutation makes sure we are always referring to the same state.